08_files_lab/01.py

```diff
@@ -1,6 +1,42 @@
-"""
-Write a program that takes 2 file names
-and appends the second file's contents to
-the end of the first one
-"""
-
+#!/usr/bin/env python
+
+"""
+This program concatinate file_1 into file_2
+
+"""
+__author__ = 'Eabel'
+
+import sys
+import os
+import argparse
+
+def is_valid_file_path(file_path):
+    if os.path.isfile(file_path):
+        return True
+    else :      
+        print "Error - invalid file path",file_path
+        return False
+
+########################################################################################################################
+#
+#
+def main():
+    parser = argparse.ArgumentParser(description='This program concatinate file_1 into file_2')
+    
+    parser.add_argument("file_1", help="path to file")
+    parser.add_argument("file_2", help="path to file")    
+    args = parser.parse_args()
+    if not is_valid_file_path(args.file_1) or not is_valid_file_path(args.file_2):

```

Comment:
Prefer using try/except over if for detecting errors. This saves you some extra tests and usually results in more correct code. For example - consider what happens if file_2 is not writable.

try/except gives you one place to handle everything that can break, and allows handling different errors differently